File: app/src/main/java/com/lavish/toprestro/models/Restaurant.java
package com.lavish.toprestro.models;

public class Restaurant {

    //TODO : Image upload & compress

    public String id, name, imageURL, ownerEmail;
    public float avgRating;
    public int noOfRatings;

    public Restaurant() {
    }

    public Restaurant(String name, String ownerEmail) {
        this.name = name;
        this.ownerEmail = ownerEmail;
    }
}

File: app/src/main/java/com/lavish/toprestro/models/Restaurant.kt
package com.lavish.toprestro.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.lavish.toprestro.other.RESTAURANTS_TABLE

@Entity(tableName = RESTAURANTS_TABLE)
data class Restaurant(
        @PrimaryKey
        var rId: String = "",
        var id: String? = null,
        var name: String? = null,
        var imageURL: String? = null,
        var ownerEmail: String? = null,
        var avgRating: Float = 0f,
        var noOfRatings: Int = 0) {

    constructor() : this("")

    override fun toString(): String {
        return "Restaurant(name=$name)"
    }


}